### ISSUE TITLE
🚀 Shortening of hunk indentation

### ISSUE THREAD

**Issue by Prince-Mandor** [dandavison/delta#1224]
Line wrapping does not work for normal or blame view for now, so I come with other idea of shortening code lines.
Is it possible to find shortest sequence of leading spaces in each hunk, and (if it is longer than 1 or 2) replace it with some character like "⇥" or any other? 
For most situation diffed parts is somewhere in the middle of code and indented, but this indentation is mostly useless from perspective of viewing changes
```
30:30|      while (count) {
31:31|        ssize_t rv = kernel_write(file, p, count, pos);
32:32|        if (rv < 0) { 
33:  |          if (rv == -EINTR)
  :33|          if (rv == -EINTR || rv == -EAGAIN)
34:34|            continue;
```
will look like 
```
30:30|⇥while (count) {
31:31|⇥  ssize_t rv = kernel_write(file, p, count, pos);
32:32|⇥  if (rv < 0) { 
33:  |⇥    if (rv == -EINTR)
  :33|⇥    if (rv == -EINTR || rv == -EAGAIN)
34:34|⇥      continue;
```
While working with narrow columns (like in side-by-side view or in preview of fzf), every character counts. And removing even two useless spaces will be useful, and especially twelve

**Comment by dandavison** [comment]
Hi @Prince-Mandor, that's an interesting idea.

One initial thing that comes to mind: does it bother you that successive hunks in the same function will have inconsistent alignment? E.g. below, the last line of the second hunk is the expression returned from the function whose definition is seen in the first hunk, and it's nice that it is indented in the way you'd expect relative to the `def` line. (I'm not arguing against the idea! And of course, it would be opt-in, not default.)


<table>
    <tbody>
        <tr>
            <td><img width="700" alt="image" src="https://user-images.githubusercontent.com/52205/198295972-e52d588c-59e4-44c4-bccc-7da46842d8d3.png">
</td>
        </tr>
    </tbody>
  </table>

**Comment by Prince-Mandor** [comment]
> Hi @Prince-Mandor, that's an interesting idea.

Hi! And thank you for great git pager!

> One initial thing that comes to mind: does it bother you that successive hunks in the same function will have inconsistent alignment? E.g. below, the last line of the second hunk is the expression returned from the function whose definition is seen in the first hunk, and it's nice that it is indented in the way you'd expect relative to the `def` line. 

I even think, this is misleading. Imagine same situation not on line 1 and 8, but on line 132 and 147, so numbers doesn't bring to mind real file content. Indentation looks similar, but in reality there are couple of wrongly-indented _if_s in between and we do not know really, is it same level or even same function (especially in languages with function overload, where "same name" doesn't mean "same function").
For me, proper way of thinking in such situation: 'I don't know'. It is some indentation from something not visible, just by accident on same level as previous hunk

> (I'm not arguing against the idea! And of course, it would be opt-in, not default.)

Of course

**Comment by dandavison** [comment]
> For me, proper way of thinking in such situation: 'I don't know'. It is some indentation from something not visible, just by accident on same level as previous hunk

Right, I agree that this viewpoint is totally reasonable and technically correct.

Is this something you'd be interested in implementing?

(FWIW, personally I always use my terminal emulator in full screen mode, never split horizontally into more than two terminals, and side-by-side has wrapping, so I think I might not use it myself. But I can imagine it being useful for people with different work styles, and I have played around briefly with fzf previews involving delta so I know they are an interesting source of possibilities.)
